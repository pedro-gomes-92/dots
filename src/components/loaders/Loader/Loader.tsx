import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseLoader, BaseLoaderProps } from '../BaseLoader';

export interface LoaderProps extends BaseLoaderProps {}

export const Loader: FunctionComponent<LoaderProps> = ({ className, loader, ...rest }: LoaderProps): JSX.Element => (
  <BaseLoader className={classnames('is-loader', className)} {...rest}>
    {loader}
  </BaseLoader>
);
