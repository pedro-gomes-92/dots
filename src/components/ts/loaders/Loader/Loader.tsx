import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { IconLoader } from '../IconLoader';
import { BaseLoader, BaseLoaderTypeProps } from '../BaseLoader';

export interface LoaderProps extends Omit<BaseLoaderTypeProps, 'children'> {}

export const Loader: FunctionComponent<LoaderProps> = ({
  className,
  sizeIcon,
  icon,
  ...rest
}: LoaderProps): JSX.Element => (
  <BaseLoader className={classnames('is-loader', className)} {...rest}>
    <IconLoader name={icon} size={sizeIcon} />
  </BaseLoader>
);
