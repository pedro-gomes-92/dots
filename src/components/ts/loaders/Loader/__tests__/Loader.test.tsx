import React from 'react';
import { render } from '@testing-library/react';

import { Loader } from '../Loader';

describe('Given <Loader />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<Loader />);
      expect(container).toMatchSnapshot();
    });
  });
});
