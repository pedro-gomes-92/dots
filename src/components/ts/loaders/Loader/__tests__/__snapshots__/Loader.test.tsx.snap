// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Loader /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="loaders is-loader"
  >
    <span
      class="is-animated is-animated-rotateRight is-animated-slow is-animated-infinite icon is-loader is-relative"
    >
      <i
        class="icon-font icon-font-loader"
      />
    </span>
  </div>
</div>
`;
